tests/unit/windows/TestUwbAppConfiguration.cxx

```diff
@@ -114,8 +135,316 @@ TEST_CASE("UwbAppConfiguration performs allocation for contained value correctly
+
+        std::vector<std::unique_ptr<IUwbAppConfigurationParameter>> parameters;
+
+        for (const auto& [parameterTag, parameterValueVariant] : sessionUwbMap) {
+            const auto ddiType = windows::devices::detail::AppConfigUwbConfigurationTagMap.at(parameterTag);
+            REQUIRE(ddiType == param.paramType);
+
+            parameters.push_back(std::make_unique<UwbAppConfigurationParameter<uint16_t>>(slotDurationExpected, parameterTag));
+
+            UwbSetAppConfigurationParameters testStruct{ parameters, sessionData.sessionId };
+
+            const auto& ddiParams = testStruct.DdiParameters();
+
+            REQUIRE(ddiParams.sessionId == sessionData.sessionId);
+            REQUIRE(ddiParams.appConfigParamsCount == std::size(parameters));
+            REQUIRE(ddiParams.size == parameters[0]->DdiSize() + offsetof(UWB_SET_APP_CONFIG_PARAMS, appConfigParams[0]));
+            REQUIRE(std::memcmp(ddiParams.appConfigParams, parameters[0]->DdiBuffer().data(), parameters[0]->DdiSize()) == 0);
+        }
+    }
+
+    SECTION("uint32_t works")
+    {
+        using namespace windows::devices;
+
+        constexpr uint32_t rangingIntervalExpected = 0xFFFF'2000;
+
+        UwbSessionData sessionData;
+        sessionData.sessionId = 1;
+        sessionData.uwbConfiguration = UwbConfiguration::Create().SetRangingInterval(rangingIntervalExpected);
+        sessionData.uwbConfigurationAvailable = true;
+
+        auto params = GenerateUwbSetAppConfigParameterDdi(sessionData);
+        const auto& ddiParams = params.DdiParameters();
+
+        REQUIRE(ddiParams.sessionId == sessionData.sessionId);
+        REQUIRE(ddiParams.appConfigParamsCount == 1);
+
+        const auto param = ddiParams.appConfigParams[0];
+        const auto sessionUwbMap = sessionData.uwbConfiguration.GetValueMap();
+
+        std::vector<std::unique_ptr<IUwbAppConfigurationParameter>> parameters;
+
+        for (const auto& [parameterTag, parameterValueVariant] : sessionUwbMap) {
+            const auto ddiType = windows::devices::detail::AppConfigUwbConfigurationTagMap.at(parameterTag);
+            REQUIRE(ddiType == param.paramType);
+
+            parameters.push_back(std::make_unique<UwbAppConfigurationParameter<uint32_t>>(rangingIntervalExpected, parameterTag));
+
+            UwbSetAppConfigurationParameters testStruct{ parameters, sessionData.sessionId };
+
+            const auto& ddiParams = testStruct.DdiParameters();
+
+            REQUIRE(ddiParams.sessionId == sessionData.sessionId);
+            REQUIRE(ddiParams.appConfigParamsCount == std::size(parameters));
+            REQUIRE(ddiParams.size == parameters[0]->DdiSize() + offsetof(UWB_SET_APP_CONFIG_PARAMS, appConfigParams[0]));
+            REQUIRE(std::memcmp(ddiParams.appConfigParams, parameters[0]->DdiBuffer().data(), parameters[0]->DdiSize()) == 0);
+        }
+    }
+
+    // TODO UWB_APP_CONFIG_PARAM_TYPE_RANGING_ROUND_CONTROL
```

Comment:
Is this still outstanding? If not, can it be removed?

Reply:
Still outstanding. I'm not sure the role of this parameter. It seems to be related to RangingMethod but I'm not sure